src/body_operations.cpp

```diff
@@ -86,19 +86,19 @@ shapes::ShapeConstPtr bodies::constructShapeFromBody(const bodies::Body* body)
   {
     case shapes::SPHERE:
     {
-      const auto& dims = static_cast<const bodies::Sphere*>(body)->getScaledDimensions();
+      const auto& dims = static_cast<const bodies::Sphere*>(body)->bodies::Sphere::getScaledDimensions();
```

Comment:
Maybe I'd add a comment explaining this unusual syntax.

```suggestion
      // The syntax used for calling getScaledDimensions() allows skipping virtual dispatch - we already know the exact type of the body.
      const auto& dims = static_cast<const bodies::Sphere*>(body)->bodies::Sphere::getScaledDimensions();
```

Reply:
I added a slightly different comment.